source/Calamari.AzureServiceFabric/Integration/AzureServiceFabricPowerShellContext.cs

```diff
@@ -57,31 +61,39 @@ public CommandResult ExecuteScript(Script script,
                 clientCertThumbprint = variables.Get($"{certificateVariable}.{CertificateVariables.Properties.Thumbprint}");
             }
+            else if (securityMode == AzureServiceFabricSecurityMode.SecureAzureAD.ToString())
+            {
+                if (string.IsNullOrWhiteSpace(aadUserCredUsername))  return new CommandResult("Failed to find a value for the AAD username.", 0);
+                if (string.IsNullOrWhiteSpace(aadUserCredPassword))  return new CommandResult("Failed to find a value for the AAD password.", 0);
+
+                var aadTokenTask = GetAzureADAccessToken(serverCertThumbprint, connectionEndpoint, aadUserCredUsername, aadUserCredPassword);
+                var aadToken = Task.Run(() => aadTokenTask).GetAwaiter().GetResult();
+
+                SetOutputVariable("OctopusFabricAadToken", aadToken);
+            }
 
             // Set output variables for our script to access.
-            SetOutputVariable("OctopusFabricConnectionEndpoint", variables.Get(SpecialVariables.Action.ServiceFabric.ConnectionEndpoint));
-            SetOutputVariable("OctopusFabricSecurityMode", variables.Get(SpecialVariables.Action.ServiceFabric.SecurityMode));
-            SetOutputVariable("OctopusFabricServerCertThumbprint", variables.Get(SpecialVariables.Action.ServiceFabric.ServerCertThumbprint));
+            SetOutputVariable("OctopusFabricConnectionEndpoint", connectionEndpoint);
+            SetOutputVariable("OctopusFabricSecurityMode", securityMode);
+            SetOutputVariable("OctopusFabricServerCertThumbprint", serverCertThumbprint);
             SetOutputVariable("OctopusFabricClientCertThumbprint", clientCertThumbprint);
             SetOutputVariable("OctopusFabricCertificateFindType", variables.Get(SpecialVariables.Action.ServiceFabric.CertificateFindType, "FindByThumbprint"));
             SetOutputVariable("OctopusFabricCertificateFindValueOverride", variables.Get(SpecialVariables.Action.ServiceFabric.CertificateFindValueOverride));
             SetOutputVariable("OctopusFabricCertificateStoreLocation", variables.Get(SpecialVariables.Action.ServiceFabric.CertificateStoreLocation, "LocalMachine"));
             SetOutputVariable("OctopusFabricCertificateStoreName", variables.Get(SpecialVariables.Action.ServiceFabric.CertificateStoreName, "MY"));
-            SetOutputVariable("OctopusFabricAadCredentialType", variables.Get(SpecialVariables.Action.ServiceFabric.AadCredentialType));
-            SetOutputVariable("OctopusFabricAadClientCredentialSecret", variables.Get(SpecialVariables.Action.ServiceFabric.AadClientCredentialSecret));
-            SetOutputVariable("OctopusFabricAadUserCredentialUsername", variables.Get(SpecialVariables.Action.ServiceFabric.AadUserCredentialUsername));
-            SetOutputVariable("OctopusFabricAadUserCredentialPassword", variables.Get(SpecialVariables.Action.ServiceFabric.AadUserCredentialPassword));
```

Comment:
Doing Azure AD token retrieval in C# means we don't need to pass these to powershell.  Note that (as mentioned in the in-code comment below), we don't support the AAD `SecureClientCertificate` auth type, mostly because (as far as I can tell) Azure doesn't support it for Service Fabric, but also because we filter it out in `AzureServiceFabricClusterEndpointValidator.cs`